app/models/fact.rb

```diff
@@ -1,8 +1,20 @@
 class Fact < ActiveRecord::Base
   belongs_to :organization
 
-  validates_presence_of :organization, :name, :variable_name, :expression
+  validates_presence_of :organization, :name, :variable_name, :expression, :simulation
   validates :variable_name,
     uniqueness: {scope: :organization_id},
     format: {with: /\A\w+\Z/}
+  validate :fact_has_values, :fact_has_no_errors
+
+  private
+
+  def fact_has_values
+    values_present = simulation && simulation["sample"] && simulation["sample"]["values"] && simulation["sample"]["values"].length > 0
```

Comment:
Also, this is a lot of &&s.  There's a gem called ``andand`` that allows you to do something like ``simulation.andand([], 'sample).andand([],'values'), but don't worry about it now.